### PR TITLE
Add support for software_token_mfa_configuration

### PR DESCRIPTION
**Pull request by jfharden**
Resolution to https://github.com/lgallard/terraform-aws-cognito-user-pool/issues/13

Add support for software TOTP mfa tokens.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -12,6 +12,9 @@ resource "aws_cognito_user_pool" "pool" {
   username_configuration {
     case_sensitive = var.case_sensitive
   }
+  software_token_mfa_configuration {
+    enabled = var.software_token_mfa_configuration_enabled
+  }
```
Comment (lgallard) [inline main.tf:17]: I would rather use a dynamic block here. Why? According to the documentation:

> # Software Token MFA Configuration
>
> The following arguments are required in the software_token_mfa_configuration configuration block:
>
> * `enabled` - (Required) Boolean whether to enable software token Multi-Factor (MFA) tokens, such as Time-based One-Time Password (TOTP). To disable software token MFA when `sms_configuration` is not present, the `mfa_configuration` argument must be set to OFF and the `software_token_mfa_configuration` configuration block must be fully removed.**

To be able to accomplish the above, I would change it to something like: 

``` 
  # software_token_mfa_configuration
  dynamic "software_token_mfa_configuration" {
    for_each = local.software_token_mfa_configuration
      content {
        enabled = lookup(software_token_mfa_configuration.value, "enabled")
       }
    }
```
And add this in locals:
```

local {
...
  # software_token_mfa_configuration
  # If no sms_configuration list is provided, build a software_token_mfa_configuration using the default values
  software_token_mfa_configuration_default = {
    enabled = lookup(var.software_token_mfa_configuration, "enabled", null) == null ? var.software_token_mfa_configuration_enabled : lookup(var.software_token_mfa_configuration, "enabled")
  }

  software_token_mfa_configuration = (length(var.sms_configuration) == 0 || sms_configuration == null) && mfa_configuration == "OFF" ? [] : [local.software_token_mfa_configuration_default]

}
```  
In the `variables.tf` I would add the `software_token_mfa_configuration` map:

```
# software_token_mfa_configuration
variable "software_token_mfa_configuration" {
  description = "Configuration block for software token Mult-Factor Authentication (MFA) "
  type        = map
  default     = {}
}

```
This way you can create the block using the variable `software_token_mfa_configuration_enabled`:


```
module "aws_cognito_user_pool_simple_example" {

   source = "../modules/terraform-aws-cognito-user-pool"
   
   ...
   
    software_token_mfa_configuration_enabled  = true
}
```  
Or by defining a block like this one:

```
module "aws_cognito_user_pool_simple_example" {

   source = "../modules/terraform-aws-cognito-user-pool"
   
   ...
   
    software_token_mfa_configuration  = {
     enabled = true
    }

```

Comment (jfharden) [reply main.tf:17]: Sure thing, I'll update in the morning. Thanks for the feedback

Comment (lgallard) [reply main.tf:17]: Great! Thanks @jfharden

Comment (jfharden) [reply main.tf]: I've made those changes (sorry it was a bit slowed than I said, our internet has been down all morning!)

**Comment by lgallard** [comment]
@jfharden great job!! Thank you for applying the changes!

**Pull request merged**